swift/Lottie.swift

```diff
@@ -0,0 +1,152 @@
+import CoreGraphics
+import CoreMedia
+
+import thorvg
+
+/// Errors that can occur while working with Lottie animations.
+public enum LottieError: Error {
+    case failedToLoadLottieFromString
+    case failedToLoadLottieFromPath
+    case failedToDrawLottieFrameOntoCanvas
+    case failedToInitialiseTvgEngine
+    case frameIndexOutOfBounds
+}
+
+/// Shorthand for the buffer type used to represent image pixel data.
+public typealias Buffer = UnsafeMutablePointer<UInt32>
+
+/// Shorthand for a pointer to an internal Lottie animation object
+private typealias Animation = OpaquePointer
+
+/// Shorthand for a pointer to an internal rendering canvas
+private typealias Canvas = OpaquePointer
+
+/// Object used to load and render Lottie frames.
+public class Lottie {
+    /// The number of frames in the Lottie animation.
+    public let numberOfFrames: Int
+
+    /// The duration of the Lottie animation.
+    public let duration: CMTime
+
+    /// The Lottie animation, used for manipulating and rendering frames.
+    private let animation: Animation?
+
+    /// The canvas used for rendering the animation frames, allowing reuse across renders.
+    private var canvas: Canvas? = nil
+
+    /// Initializes a new Lottie instance from a file path.
+    /// - Parameters:
+    ///   - path: The file path of the Lottie animation.
+    public convenience init(path: String) throws {
+        let animation = tvg_animation_new()
+        let picture = tvg_animation_get_picture(animation)
+
+        guard tvg_picture_load(picture, path) == TVG_RESULT_SUCCESS else {
+            throw LottieError.failedToLoadLottieFromPath
+        }
+
+        try self.init(animation: animation)
+    }
+
+    /// Initializes a new Lottie instance from a string representing the animation.
+    /// - Parameters:
+    ///   - string: The string representing the Lottie animation.
+    public convenience init(string: String) throws {
+        let animation = tvg_animation_new()
+        let picture = tvg_animation_get_picture(animation)
+
+        guard let cString = string.cString(using: .utf8),
+              tvg_picture_load_data(picture, cString, UInt32(cString.count), "lottie", "", false) == TVG_RESULT_SUCCESS
+        else {
+            throw LottieError.failedToLoadLottieFromString
+        }
+
+        try self.init(animation: animation)
+    }
+
+    /// Initializes the Lottie instance with frame count and duration extracted from the animation pointer.
+    private init(animation: Animation?) throws {
+        var numberOfFrames: Float = 0
+        tvg_animation_get_total_frame(animation, &numberOfFrames)
```

Comment:
Good question. 

Looking at the documentation for both `tvg_animation_get_total_frame` and `tvg_animation_get_duration` neither of them actually give us any indicator as to an error with the `Animation` instance. The only error checking those functions do is to check whether the parameters are `null` in which case it throws a `TVG_RESULT_INVALID_ARGUMENT` error. 

So even if we did catch errors across these two functions, they wouldn't give us that extra coverage we're looking for. 

I believe our error handling across the `tvg_picture_load` and `tvg_picture_load_data` functions are what is doing the real heavy-lifting here, because that's what we're really trying to ensure. That we've loaded Lottie successfully and can then start operating on it. 

IMO, error-handling across those two functions will suffice.